crates/model/src/order.rs

```diff
@@ -359,33 +360,91 @@ impl OrderData {
+  "buyToken": "0x6810e776880c02933d47db1b9fc05908e5386b96",
+  "receiver": "0x6810e776880c02933d47db1b9fc05908e5386b96",
+  "sellAmount": "1234567890",
+  "buyAmount": "1234567890",
+  "validTo": 0,
+  "feeAmount": "1234567890",
+  "kind": "buy",
+  "partiallyFillable": true,
+  "sellTokenBalance": "erc20",
+  "buyTokenBalance": "erc20",
+  "signingScheme": "eip712",
+  "signature": "0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+  "from": "0x6810e776880c02933d47db1b9fc05908e5386b96",
+  "quoteId": 0,
+  "appData": "{\"version\":\"0.9.0\",\"metadata\":{}}",
+  "appDataHash": "0x0000000000000000000000000000000000000000000000000000000000000000"
+}))]
 pub struct OrderCreation {
     // These fields are the same as in `OrderData`.
+    /// Address of token sold.
+    #[schema(value_type = String)]
     pub sell_token: H160,
+    /// Address of token bought.
+    #[schema(value_type = String)]
     pub buy_token: H160,
+    /// An optional address to receive the proceeds of the trade instead of the
+    /// `owner` (i.e. the order signer).
     #[serde(default)]
+    #[schema(value_type = Option<String>)]
     pub receiver: Option<H160>,
+    /// Amount of `sellToken` to be sold in atoms.
     #[serde_as(as = "HexOrDecimalU256")]
+    #[schema(value_type = String)]
     pub sell_amount: U256,
+    /// Amount of `buyToken` to be sold in atoms.
     #[serde_as(as = "HexOrDecimalU256")]
+    #[schema(value_type = String)]
     pub buy_amount: U256,
+    /// Unix timestamp (`uint32`) until which the order is valid.
+    #[schema(example = 0)]
     pub valid_to: u32,
+    /// feeRatio * sellAmount + minimal_fee in atoms.
     #[serde_as(as = "HexOrDecimalU256")]
+    #[schema(value_type = String)]
     pub fee_amount: U256,
+    #[schema(example = OrderKind::Buy)]
     pub kind: OrderKind,
+    /// Is the order fill-or-kill or partially fillable?
+    #[schema(example = true)]
     pub partially_fillable: bool,
     #[serde(default)]
+    #[schema(value_type = String, example = "erc20")]
     pub sell_token_balance: SellTokenSource,
     #[serde(default)]
+    #[schema(value_type = String, example = "erc20")]
     pub buy_token_balance: BuyTokenDestination,
-
+    /// If set, the backend enforces that this address matches what is decoded
+    /// as the *signer* of the signature. This helps catch errors with
+    /// invalid signature encodings as the backend might otherwise silently
+    /// work with an unexpected address that for example does not have
+    /// any balance.
+    #[schema(value_type = Option<String>)]
     pub from: Option<H160>,
+    /// If set, the backend enforces that this address matches what is decoded
+    /// as the *signer* of the signature. This helps catch errors with
+    /// invalid signature encodings as the backend might otherwise silently
+    /// work with an unexpected address that for example does not have
+    /// any balance.
```

Comment:
If you look at the generated yaml, for some reason the `flattened` types don't show up as named arguments on the top of `OrderCreation` but instead unnamed in the `allOf` definition. Any idea why that is?

Reply:
This seems to cause a bigger issue, the field attributes (ie. "example") on the flattened type seem to overwrite the parent types definition, investigating further.